src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -218,45 +281,71 @@ public IAsyncCursor<TResult> Execute(IReadBinding binding, CancellationToken can
                 throw new ArgumentException("The binding value passed to ChangeStreamOperation.ExecuteAsync must implement IReadBindingHandle.", nameof(binding));
             }
 
-            var cursor = await ResumeAsync(bindingHandle, _resumeAfter, cancellationToken).ConfigureAwait(false);
+            var cursor = await ResumeAsync(bindingHandle, cancellationToken).ConfigureAwait(false);
+            if (_startAtOperationTime == null && _resumeAfter == null)
+            {
+                _startAtOperationTime = binding.Session.OperationTime;
+            }
             return new ChangeStreamCursor<TResult>(cursor, _resultSerializer, bindingHandle.Fork(), this);
         }
 
         /// <inheritdoc />
-        public IAsyncCursor<RawBsonDocument> Resume(IReadBinding binding, BsonDocument resumeAfter, CancellationToken cancellationToken)
+        public IAsyncCursor<RawBsonDocument> Resume(IReadBinding binding, CancellationToken cancellationToken)
         {
-            var aggregateOperation = CreateAggregateOperation(resumeAfter);
```

Comment:
resumeAfter and _resumeAfter were kind of at odds with each other.

Now that we track the resume token in the operation's _resumeAfter field we just use that and there is no need for a resumeAfter parameter any more.